app/models/post.rb

```diff
@@ -1,2 +1,15 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_posts_counter
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] (Optional) It would be nice to make this function a _**private**_ one. 

```
private 

def update_posts_counter
  write code
end
```

#### When you make functions private, you prevent the function from being accessible by others and that way, you prevent malicious attacks. 👍

Reply:
@Gambit142 Thank you for your valuable feedback.